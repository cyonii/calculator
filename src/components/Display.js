import PropTypes from 'prop-types';

const Display = (props) => {
  const { result } = props;
  const displayCSS = 'px-2 py-3 py-lg-4 rounded-pill mb-3 bg-secondary text-end fs-5 text-white';

  return <div className={`${displayCSS}`}>{result}</div>;
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
