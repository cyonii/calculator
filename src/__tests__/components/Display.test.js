import { cleanup, render, screen } from '@testing-library/react';
import Display from '../../components/Display';

describe('<Display /> Component', () => {
  afterEach(() => cleanup);

  it('should display correctly', () => {
    render(<Display total="" next="" operator="" />);

    const display = screen.getByTestId('display');
    expect(display).toBeInTheDocument();
    expect(display).toHaveTextContent('0');
  });

  it('should display `total` value when `next` value is not given', () => {
    render(<Display total="234" next="" operator="" />);

    const display = screen.getByTestId('display');
    expect(display).toHaveTextContent('234');
  });

  it('should NOT display `next` value when `operator` value is NOT given', () => {
    render(<Display total="234" next="64425" operator="" />);

    const display = screen.getByTestId('display');
    expect(display).not.toHaveTextContent('64425');
    expect(display).toHaveTextContent('234');
  });

  it('should display `next` value when `next` & `operator` is given', () => {
    render(<Display total="234" next="43567" operator="+" />);

    const display = screen.getByTestId('display');
    expect(display).toHaveTextContent('43567');
  });

  it('should match snapshot', () => {
    render(<Display total="234" next="43567" operator="+" />);

    const display = screen.getByTestId('display');
    expect(display).toMatchSnapshot();
  });
});
