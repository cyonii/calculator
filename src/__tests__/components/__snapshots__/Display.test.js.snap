// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> Component should match snapshot 1`] = `
<div
  class="px-2 py-3 py-lg-4 rounded-pill mb-3 bg-secondary text-end fs-5 text-white"
  data-testid="display"
>
  43567
</div>
`;
